flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/stream/sql/ProcessTableFunctionTest.java

```diff
@@ -0,0 +1,326 @@
+    }
+
+    @Test
+    void testReservedArg() {
+        util.addTemporarySystemFunction("f", ReservedArgFunction.class);
+        assertThatThrownBy(() -> util.verifyRelPlan("SELECT * FROM f(uid => 'my-ptf')"))
+                .hasRootCauseMessage(
+                        "Function signature must not declare system arguments. Reserved argument names are: [uid]");
+    }
+
+    @Test
+    void testInvalidMultiTable() {
+        util.addTemporarySystemFunction("f", MultiTableFunction.class);
+        assertThatThrownBy(
+                        () -> util.verifyRelPlan("SELECT * FROM f(r1 => TABLE t1, r2 => TABLE t1)"))
+                .hasRootCauseMessage(
+                        "Currently, only signatures with at most one table argument are supported.");
+    }
+
+    @Test
+    void testInvalidRowInsteadOfTable() {
+        util.addTemporarySystemFunction("f", TableAsRowFunction.class);
+        assertThatThrownBy(() -> util.verifyRelPlan("SELECT * FROM f(r => ROW(42), i => 1)"))
+                .hasRootCauseMessage(
+                        "Invalid argument value. Argument 'r' expects a table to be passed.");
+    }
+
+    @Test
+    void testInvalidSetSemantics() {
+        util.addTemporarySystemFunction("f", TableAsRowFunction.class);
+        assertThatThrownBy(
+                        () ->
+                                util.verifyRelPlan(
+                                        "SELECT * FROM f(r => TABLE t1 PARTITION BY name, i => 1)"))
+                .hasRootCauseMessage(
+                        "Only tables with set semantics may be partitioned. "
+                                + "Invalid PARTITION BY clause in the 0-th operand of table function 'f'");
+    }
+
+    @Test
+    void testInvalidPartitionByClause() {
+        util.addTemporarySystemFunction("f", TableAsSetFunction.class);
+        assertThatThrownBy(
+                        () ->
+                                util.verifyRelPlan(
+                                        "SELECT * FROM f(r => TABLE t1 PARTITION BY invalid, i => 1)"))
+                .hasRootCauseMessage(
+                        "Invalid column 'invalid' for PARTITION BY clause. Available columns are: [name, score]");
+    }
+
+    @Test
+    void testUnsupportedOrderByClause() {
+        util.addTemporarySystemFunction("f", TableAsSetFunction.class);
+        assertThatThrownBy(
+                        () ->
+                                util.verifyRelPlan(
+                                        "SELECT * FROM f(r => TABLE t1 PARTITION BY name ORDER BY score, i => 1)"))
+                .hasRootCauseMessage("ORDER BY clause is currently not supported.");
+    }
+
+    private void assertReachesOptimizer(String sql) {
+        assertThatThrownBy(() -> util.verifyRelPlan(sql))
+                .hasMessageContaining(
+                        "This exception indicates that the query uses an unsupported SQL feature.");
+    }
+
+    /** Testing function. */
```

Comment:
I guess instead of this repeating comment we could just add one like 
```java
// -------------------------------------------
// Testing functions
// -------------------------------------------
```
or something like that
WDYT?